### PR TITLE
Initial mobile client setup

### PR DESCRIPTION
**Pull request by alan-hadyk**
# Remarks

1. During code review, please ignore the following auto-generated files and folders:
- `client-mobile/ios` (build with iOS version)
- `client-mobile/src/api/schema.ts` (API Schema)

2. Certain files and components are dummy code (usually marked with comment similar to `TODO - Dummy`).

# Testing scenario 1

1. Follow `Prerequisites` and `Development - iOS simulator` instructions in `client-mobile/README.md`
2. Make sure that the app runs on the iOS simulator. Try to move around, switch between tabs and use the API (via black buttons).

Zrzut ekranu 2023-10-8 o 19 28 15

3. Try to run E2E tests by following `E2E tests` section in `client-mobile/README.md`

# Testing scenario 2

1. Follow `Prerequisites` and `Development - Expo app (iOS)` instructions in `client-mobile/README.md`
2. Make sure that the app runs on the actual mobile device via `Expo Go` app. Try to move around, switch between tabs and use the API (via black buttons).

### CODE REVIEW

File: client-mobile/README.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+1.Create `client-mobile/.env` file. Inside, paste the content from `client-mobile/.env.example`.
+
+2.[Install Xcode](https://developer.apple.com/xcode/)
+
+3.Run Expo in development mode with iOS simulator
+
+```bash
+yarn ios:dev
+```
```
[the remark this answers is not in the document]
Comment (alan-hadyk) [reply client-mobile/README.md:32]: Added, along with additional instructions for `Development - Expo app in iOS simulator`

[the remark this answers is not in the document]
Comment (alan-hadyk) [reply client-mobile/README.md:32]: Also added Table of contents:

Zrzut ekranu 2023-10-13 o 13 51 44

File: client-mobile/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    "import/no-self-import": 2,
+    "import/no-unresolved": 0,
+    "no-multiple-empty-lines": [
+      "error",
+      {
+        max: 1,
+        maxEOF: 0,
+      },
+    ],
+    "no-unused-vars": "off",
```
[the remark this answers is not in the document]
Comment (alan-hadyk) [reply client-mobile/.eslintrc.js:49]: https://typescript-eslint.io/rules/no-unused-vars/#how-to-use

Zrzut ekranu 2023-10-13 o 13 54 42

File: client-mobile/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alan-hadyk) [reply client-mobile/README.md:32]: We're gonna leave this for now. Created an issue - [https://github.com/alan-hadyk/grocify/issues/5](https://github.com/alan-hadyk/grocify/issues/5)

File: client-mobile/src/helpers/functions/@types/composeFunctions.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface ComponentEnhancer<TInner, TOutter> {
```
[the remark this answers is not in the document]
Comment (alan-hadyk) [reply client-mobile/src/helpers/functions/@types/composeFunctions.ts:1]: Changed to `TOuter`

**Review by angulawk — APPROVED** [review]

**Pull request merged**